test/tc_client_tests.cpp

```diff
@@ -478,14 +479,18 @@ TEST(TASK_CONTROLLER_CLIENT_TESTS, MessageEncoding)
 
 	CANHardwareInterface::stop();
 	CANHardwareInterface::set_number_of_can_channels(0);
+
+	//! @todo try to reduce the reference count, such that that we don't use a control function after it is destroyed
+	ASSERT_TRUE(tcPartner->destroy(3));
+	ASSERT_TRUE(internalECU->destroy(3));
```

Comment:
As a note, we might want to make the classes outside of the CANManager use the `weak_ptr` variant of the control functions sometime. Currently they are not aware of when a ControlFunction is destroyed/deleted. Though such a change isn't really disruptive, since we should be able to swap a `shared_ptr` (in most places) with a `weak_ptr` without any changes required at the application level, hence the priority on this is not that high I think.